docs/releases/2.2.txt

```diff
@@ -57,6 +57,18 @@ Minor features
 
 * The ``HttpRequest`` is now passed as the first positional argument to
   :meth:`.RemoteUserBackend.configure_user`, if it accepts it.
+* Permissions for :ref:`proxy models <proxy-models>` will now be created using
+  the content type of the proxy model, rather than, in previous versions, the
+  content type of the underlying concrete model.
+
+  This means that users granted permissions for proxy models having a different
+  ``app_label`` than their concrete models will now be able to correctly access
+  corresponding admin pages. (In this respect, the change is transparent for
+  proxy models having the same ``app_label`` as their concrete model.)
+
+  A reversible migration is provided to automatically update existing
+  permissions when running :djadmin:`migrate`.  Reviewing admin permissions for
```

Comment:
Reviewing the permissions in what way?

Reply:
This is related to:
>This means that users granted permissions for proxy models having a different ``app_label`` than their concrete models will now be able to correctly access corresponding admin pages.

We want to make sure they audit the list of users/groups having such permissions prior to running the migration to avoid surprises.

Reply:
I think the documentation should describe the audit process in more detail. Perhaps an example would be helpful.

Reply:
Something like the following?
```python
from django.contrib.contenttypes import ContentType
from django.db.models import Q

from myapp.models import MyProxyModel


opts = MyProxyModel._meta

default_permissions_codenames = [
    '%s_%s' % (action, opts.model_name)
    for action in opts.default_permissions
]

permissions_query = Q(codename__in=proxy_default_permissions_codenames)
for codename, name in opts.permissions:
    permissions_query = permissions_query | Q(codename=codename, name=name)

content_type = ContentType.objects.get_for_model(MyProxyModel)

permission_queryset = Permission.objects.filter(
    permissions_query,
    content_type=concrete_content_type,
)

for permission in permission_queryset:
    print("Permission: %s" % permission.codename)

    print("Groups")
    print(permission.group_set.all())

    print("Users")
    print(permission.user_set.all())
```

Reply:
Not necessarily code. I meant describing how someone would audit and what sort of change you might make as a result.

Reply:
I think we can re-order a bit the existing text to make it more clear:

```
Permissions for proxy models
----------------------------

Permissions for proxy models are now created using the content type of the
proxy model rather than the content type of the  concrete model.

A migration is provided to update existing permissions when running
:djadmin:`migrate`.

In this respect, the change is transparent for proxy models having the same
``app_label`` as their concrete model.

However, users granted permissions for proxy models having a different
``app_label`` than their concrete models will now be able to correctly
access corresponding admin pages.  Therefore, auditing `groups` and
`users` having admin permissions (e.g. For `ProxyModel`:
`add_proxymodel`, `view_proxymodel`, `change_proxymodel`,
`delete_proxymodel`) for such proxy models prior to running
:djadmin:`migrate` is recommended. You may remove the permission
from the list of group or user permissions to prevent any unforeseen
access leak and carefully re-assign them once the :djadmin:`migrate`
command has been run.
```

Do you think that'd be enough or do you want more details about **how** to audit?

Reply:
🤔 It's not really an "access leak" but more a conscious choice of "fixing" the permissions and letting users get access to the relevant admin pages **automatically**. Let me try to rephrase one more time...

```
Permissions for proxy models
----------------------------

Permissions for proxy models are now created using the content type of the
proxy model rather than the content type of the  concrete model.

A migration is provided to update existing permissions when running
:djadmin:`migrate`.

In this respect, the change is transparent for proxy models having the same
``app_label`` as their concrete model.

However, users granted permissions for proxy models having a different
``app_label`` than their concrete models will now be able to correctly
access corresponding admin pages.  Auditing permissions of groups and
users for such proxy models (``[add|view|change|delete]_myproxy``)
prior to running :djadmin:`migrate` is recommended as you might want to
notify affected users of the changes to their admin homepage beforehand.
If you want to automatically give access to admin pages of such proxy
models, you can remove their permissions from groups and users, and
re-assign them after having run :djadmin:`migrate`.
```

Reply:
It's a typo, I meant "If you **don't** want to ...".

Reply:
Thanks, that was my assumption. I removed that sentence since it seems obvious.